src/snakeGame/Snake.java

```diff
@@ -0,0 +1,131 @@
+package snakeGame;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import edu.macalester.graphics.CanvasWindow;
+import edu.macalester.graphics.Ellipse;
+import edu.macalester.graphics.GraphicsGroup;
+import edu.macalester.graphics.GraphicsObject;
+import edu.macalester.graphics.Point;
+
+import java.awt.Color;
+
+/**
+ * Creates and manages movement of snake
+ */
+public class Snake {
+    private static final double BODY_WIDTH = 15;
+
+    private Head head;
+    private List<GraphicsObject> body;
+    private Color color;
+    private GraphicsGroup parentGroup;
+    private int bodyCollisionRadius;
+
+    /**
+     * Creates snake with head and 25 body parts
+     * @param color
+     * @param parentGroup
+     */
+    public Snake(Color color, GraphicsGroup parentGroup) {
+        head = new Head(color);
+        body = new ArrayList<>();
+        parentGroup.add(head);
+
+        this.color = color;
+        this.parentGroup = parentGroup;
+        makeLonger(25);
+    }
+
+    public List<GraphicsObject> getBodyGraphics() {
+        return body;
+    }
+
+    public Head getHead() {
+        return head;
+    }
+
+    /**
+     * Checks if snake has passed canvas bounds
+     * @return true when out of bounds
+     * @return false when in bounds
+     */
+    public boolean wallCollision(double canvasWidth, double canvasHeight) {
+        if (head.getShape().getX() < 0 || head.getShape().getX() + 30 > canvasWidth 
+        || head.getShape().getY() < 0 || head.getShape().getY() + 30 > canvasHeight) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Checks if snake head has hit snake body
+     * @return true if snake head is in contact with body
+     * @return false if no contact between head and body
+     */
+    public boolean bodyCollision() {
+        bodyCollisionRadius = 0;
+        double collisionDistance = (BODY_WIDTH/2) + 10;
+        for (GraphicsObject part : body) {
+            if (bodyCollisionRadius > 25) {
```

Comment:
Would be better to use the head size or…something instead of the magic number 25.